src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -71,6 +71,12 @@ public virtual void AddCompilationRoots(IRootingServiceProvider rootProvider)
 
             AddWellKnownTypes(rootProvider);
             AddReflectionInitializationCode(rootProvider);
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
+                var module = _typeSystemContext.GetModuleFromPath(inputFile.Value);
+                if (module.Equals(_typeSystemContext.SystemModule)) continue;
```

Comment:
Delete the `AddCompilationRootsForExports((EcmaModule)_typeSystemContext.SystemModule, rootProvider);` line in `SingleFileCompilationModuleGroup` and delete this line too.